hoomd/hdf5.py

```diff
@@ -358,3 +359,58 @@ def _write_header(self, f):
                             hoomd.context.current.device.cpp_msg.error("Quantity position is not vacant.")
                             raise RuntimeError("Error updating quantities with log_hdf5.")
                     data_set.attrs[quantities[i]] = i
+
+
+# For the HDF5 logger, I inherit from the GSDLogWriter.
+# This makes the assumption that the GSDLogWriter creates with the log function
+# a dictionary that has the names as keys and np.arrays as values.
+# HDF5 has the restriction that the dimension of the matrices cannot change.
+# This translates to the restriction that the dimensions of the logged quantities cannot change during a simulation run.
+class HDF5LogWriter(hoomd.dump.GSDLogWriter):
+    def __init__(self,logger):
+        super().__init__(logger)
+
+    def _write_frame(self,h5file):
+
+        f = None
+        if hoomd.context.current.device.comm.rank == 0:
+        f = self.h5file
```

Comment:
Where are we setting the `h5file` property?

Reply:
Also this should be indented.

Reply:
I changed it now, that the h5file file handle is initialized with the constructor.
That resulted from my confusion. The GSDLogWriter also accepted the GSDDumpWriter c++ backend in the function.

indention fixed.